tests/kafkatest/tests/core/security_rolling_upgrade_test.py

```diff
@@ -96,10 +96,15 @@ def roll_in_sasl_mechanism(self, security_protocol, new_sasl_mechanism):
         self.set_authorizer_and_bounce(security_protocol, security_protocol, KafkaService.SIMPLE_AUTHORIZER)
 
     def add_separate_broker_listener(self, broker_security_protocol, broker_sasl_mechanism):
+        # Enable the new internal listener on all brokers first
+        self.kafka.open_port(self.kafka.INTERBROKER_LISTENER_NAME)
+        self.kafka.port_mappings[self.kafka.INTERBROKER_LISTENER_NAME].security_protocol = broker_security_protocol
+        self.kafka.client_sasl_mechanism = broker_sasl_mechanism
```

Comment:
Can you elaborate why we added the two lines above?

Reply:
The Kafka service used for system test starts up a client listener and an inter-broker listener and doesn't properly support other listeners. The security protocol and SASL mechanism used for the listeners is derived from the security protocol and the mechanism specified for client/inter-broker. The two lines above work around that limitation by overriding the security protocol and SASL mechanism to allow broker to start up a listener before using it as the inter-broker listener. It would be a much larger change to add proper support for additional listeners to the Kafka service, so I just updated this test which fails very frequently at the moment.

Reply:
Thanks for the explanation. Makes sense. What is the reason why the test started failing frequently?

Reply:
Test seems to have been failing since Jan 15th. Not sure if it was more stable before that. Consumer was timing out in the logs I looked at. Will investigate more to see if there is any other issue causing the failures now.